modules/assim.sequential/R/sda.enkf_refactored.R

```diff
@@ -140,24 +158,51 @@ sda.enkf <- function(settings, obs.mean, obs.cov, Q = NULL, restart=F,
 
   ###-------------------------------------------------------------------###
   ### If this is a restart - Picking up were we left last time          ###
-  ###-------------------------------------------------------------------###----   
-  if (restart){
-    if(!file.exists(file.path(settings$outdir,"SDA", "sda.output.Rdata"))) PEcAn.logger::logger.severe("The SDA output from the older simulation doesn't exist.")
-    load(file.path(settings$outdir,"SDA", "sda.output.Rdata"))
+  ###-------------------------------------------------------------------###   
+  if (control$debug)
+    browser()
+  if (!is.null(restart)) {
+    if (!file.exists(file.path(restart, "SDA", "sda.output.Rdata")))
+      PEcAn.logger::logger.severe("The SDA output from the older simulation doesn't exist.")
+    load(file.path(restart, "SDA", "sda.output.Rdata"))
+    
+    #this is where the old simulation will be moved to
+    old.dir <-
+      lubridate::ymd_hms(names(Viz.output[[3]]) %>% tail(1)) %>% as.character()
     #--- Updating the nt and etc
-    if(!dir.exists(file.path(settings$outdir,"SDA",assimyears[t]))) dir.create(file.path(settings$outdir,"SDA",assimyears[t]))
-    # finding/moving files to it's end year dir
-    files.last.sda<-list.files.nodir(file.path(settings$outdir,"SDA"))
-    #copying
-    file.copy(file.path(file.path(settings$outdir,"SDA"),files.last.sda),
-              file.path(file.path(settings$outdir,"SDA"),paste0(assimyears[t],"/",files.last.sda))
-    )
+    if (!dir.exists(file.path(settings$outdir, "SDA", old.dir)))
+      dir.create(file.path(settings$outdir, "SDA", old.dir))
+    #copying-----
+    #SDA
+    file.copy(file.path(restart, "SDA", "sda.output.Rdata"),
+              file.path(file.path(settings$outdir, "SDA", old.dir)))
+    #run folder
+    if (!dir.exists(file.path(settings$outdir, "run")))
+      dir.create(file.path(settings$outdir, "run"))
+    file.copy(file.path(restart, "run"),
+              file.path(file.path(settings$outdir)),
+              recursive = TRUE)
+    
+    
+    params <- new.params
+    sim.time <-
+      2:nt # if It's restart I added +1 from the start to nt (which is the last year of old sim) to make the first sim in restart time t=2
```

Comment:
I'd like to discuss this in person with you because I think we have multiple options here. What I did for the single site SDA was to just keep the very last round of simulations/SDA from the previous simulation not the whole thing. But for multisite I brought the whole past history. 
We can discuss the pros and the cons of each approach next week and merge them into one.

Reply:
Agree this needs to be generalized, but I think we can merge this in first and then generalize later (e.g. when resolving conflicts with @araiho other PR)